ceph.md

```diff
@@ -0,0 +1,245 @@
+```
+for P in vms volumes images; do 
+  cephadm shell -- ceph osd pool create $P;
+  cephadm shell -- ceph osd pool application enable $P rbd;
+done
+```
+
+Create a cephx key which OpenStack can use to access the pools.
+```
+cephadm shell -- \
+   ceph auth add client.openstack \ 
+     mgr 'allow *' \
+	 mon 'allow r' \
+	 osd 'allow class-read object_prefix rbd_children, allow rwx pool=vms, allow rwx pool=volumes, allow rwx pool=images'
+```
+
+Export the cephx key and Ceph configuration file.
+```
+cephadm shell -- ceph auth get client.openstack > /etc/ceph/ceph.client.openstack.keyring
+cephadm shell -- ceph config generate-minimal-conf > /etc/ceph/ceph.conf
+```
+
+## Create a Ceph Secret
+
+Transfer the cephx key and Ceph configuration file from the previous
+section to a host which can create resources in the openstack
+namespace. Base64 encode these files and store them in two variables.
+```
+KEY=$(cat /etc/ceph/ceph.client.openstack.keyring | base64 -w 0)
+CONF=$(cat /etc/ceph/ceph.conf | base64 -w 0)
+```
+Use the variables to create a `ceph-conf-files` secret.
+```
+cat <<EOF > ceph_secret.yaml
+apiVersion: v1
+data:
+  ceph.client.openstack.keyring: $KEY
+  ceph.conf: $CONF
+kind: Secret
+metadata:
+  name: ceph-conf-files
+  namespace: openstack
+type: Opaque
+EOF
+
+oc create -f ceph_secret.yaml
+```
+
+The Ceph FSID can be extracted from the secret using the following
+command.
+```
+FSID=$(oc get secret ceph-conf-files -o json | jq -r '.data."ceph.conf"' | base64 -d | grep fsid | sed -e 's/fsid = //')
+```
+The above will be useful for configuration snippets covered later in
+the document.
+
+## Access the Ceph Secret via extraMounts
+
+Use
+[extraMounts](https://github.com/openstack-k8s-operators/lib-common/commit/c3752c8a43423c755adb16aebf644c304befc072)
+in CRs for pods which need to access the Ceph
+secret. For example, the sample
+[core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml](https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane_network_isolation_ceph.yaml)
+has the following in the `OpenStackControlPlane` CR so that the
+Glance and Cinder volume pods mount the cephx key and Ceph
+configuration files in /etc/ceph.
+
+```
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+spec:
+  extraMounts:
+    - name: v1
+      region: r1
+      extraVol:
+        - propagation:
+          - CinderVolume
+          - GlanceAPI
+          extraVolType: Ceph
+          volumes:
+          - name: ceph
+            projected:
+              sources:
+              - secret:
+                  name: ceph-conf-files
+          mounts:
+          - name: ceph
+            mountPath: "/etc/ceph"
+            readOnly: true
+```
+The `OpenStackDataPlane` can also use `extraMounts`.
+```
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlane
+spec:
+  roles:
+    edpm-compute:
+	  nodeTemplate:
+        extraMounts:
```

Comment:
Is this an indentation error or just the github view got confused?

Reply:
Thanks for catching. Will be fixed in next PS.

Details: `nodeTemplate` had "<tab><space><space>" while all of the other strings in the yaml block had N spaces. My next revision will have 6 spaces in front of `nodeTemplate`.